# HG changeset patch
# User Colin Macdonald <cbm@m.fsf.org>
# Date 1468391616 25200
#      Tue Jul 12 23:33:36 2016 -0700
# Node ID b188be2b962beb8e8d0b7079da37b1f6c76f3a3a
# Parent  2390079a8aed3d6b807772164e0bc0406ab36b9a
Call update_transform on camera property change (bug #44503)

* graphics.in.h: Add update functions for camera properties.

diff --git a/libinterp/corefcn/graphics.in.h b/libinterp/corefcn/graphics.in.h
--- a/libinterp/corefcn/graphics.in.h
+++ b/libinterp/corefcn/graphics.in.h
@@ -3594,24 +3594,24 @@ public:
 
     BEGIN_PROPERTIES (axes)
       radio_property activepositionproperty , "{outerposition}|position"
       row_vector_property alim m , default_lim ()
       radio_property alimmode , "{auto}|manual"
       color_property ambientlightcolor , color_values (1, 1, 1)
       bool_property box u , "off"
       radio_property boxstyle , "{back}|full"
-      row_vector_property cameraposition m , Matrix (1, 3, 0.0)
-      radio_property camerapositionmode , "{auto}|manual"
-      row_vector_property cameratarget m , Matrix (1, 3, 0.0)
-      radio_property cameratargetmode , "{auto}|manual"
-      row_vector_property cameraupvector m , Matrix (1, 3, 0.0)
-      radio_property cameraupvectormode , "{auto}|manual"
-      double_property cameraviewangle m , 6.6086
-      radio_property cameraviewanglemode , "{auto}|manual"
+      row_vector_property cameraposition mu , Matrix (1, 3, 0.0)
+      radio_property camerapositionmode u , "{auto}|manual"
+      row_vector_property cameratarget mu , Matrix (1, 3, 0.0)
+      radio_property cameratargetmode u , "{auto}|manual"
+      row_vector_property cameraupvector mu , Matrix (1, 3, 0.0)
+      radio_property cameraupvectormode u , "{auto}|manual"
+      double_property cameraviewangle mu , 6.6086
+      radio_property cameraviewanglemode u , "{auto}|manual"
       row_vector_property clim m , default_lim ()
       radio_property climmode al , "{auto}|manual"
       radio_property clippingstyle , "{3dbox}|rectangle"
       color_property color , color_property (color_values (1, 1, 1), radio_values ("none"))
       array_property colormap sg , Matrix ()
       array_property colororder , default_colororder ()
       double_property colororderindex , 1.0
       array_property currentpoint , Matrix (2, 3, 0.0)
@@ -3768,16 +3768,43 @@ public:
 
     void update_ycolor (void)
     { update_label_color (ylabel, ycolor); }
 
     void update_zcolor (void)
     { update_label_color (zlabel, zcolor); }
 
     void update_view (void) { sync_positions (); }
+
+    void update_cameraposition (void) { update_transform (); }
+    void update_cameratarget (void) { update_transform (); }
+    void update_cameraupvector (void) { update_transform (); }
+    void update_cameraviewangle (void) { update_transform (); }
+
+    void update_camerapositionmode (void)
+    {
+      if (camerapositionmode_is ("auto"))
+        update_cameraposition ();
+    }
+    void update_cameratargetmode (void)
+    {
+      if (cameratargetmode_is ("auto"))
+        update_cameratarget ();
+    }
+    void update_cameraupvectormode (void)
+    {
+      if (cameraupvectormode_is ("auto"))
+        update_cameraupvector ();
+    }
+    void update_cameraviewanglemode (void)
+    {
+      if (cameraviewanglemode_is ("auto"))
+        update_cameraviewangle ();
+    }
+
     void update_dataaspectratio (void) { sync_positions (); }
     void update_dataaspectratiomode (void) { sync_positions (); }
     void update_plotboxaspectratio (void) { sync_positions (); }
     void update_plotboxaspectratiomode (void) { sync_positions (); }
 
     void update_layer (void) { update_axes_layout (); }
     void update_box (void)
     {
